python/chemiscope/explore/__init__.py

```diff
@@ -1,22 +1,73 @@
+import warnings
+
 from ..jupyter import show
+from ..structures import extract_properties
 from ._metatomic import metatomic_featurizer
 
 
-__all__ = [
-    "explore",
-    "metatomic_featurizer",
-]
+__all__ = ["explore", "metatomic_featurizer", "get_featurizer"]
+
+FEATURIZER_VERSION_MAP = {"pet-mad-1.0": "1.0.0"}
+
+
+def get_featurizer(name, device=None, batch_size=1):
+    """
+    Get a featurizer by name for feature extraction. Currently available version is:
+    "pet-mad-1.0", which returns an instance of `PETMADFeaturizer
+    <https://arxiv.org/abs/2506.19674>`_.
+
+    :param str name: name of the featurizer. Must match one of the known versions.
+        Currently available is "pet-mad-1.0"
+
+    :param str device: optional. device to run the featurizer on (e.g., "cpu" or "cuda")
+
+    :param int batch_size: optional. number of structures to process per batch during
+        featurization, defaults to 1
+
+    .. warning::
+
+        Requires additional dependencies. Install it using:
+
+        .. code:: bash
+
+            pip install chemiscope[explore]
+    """
+    try:
+        from pet_mad.explore import PETMADFeaturizer
+    except ImportError as e:
+        raise ImportError(
+            f"Required package not found: {e}. Please install the "
+            "dependencies with `pip install chemiscope[explore]`."
+        )
+
+    if not isinstance(name, str):
+        raise TypeError(f"Featurizer name must be a string, got {type(name)}")
+
+    versions = FEATURIZER_VERSION_MAP.keys()
+    if name not in FEATURIZER_VERSION_MAP:
+        raise ValueError(
+            f"The featurizer version {name} does not exist. Available options are: "
+            f"{', '.join(versions) if len(versions > 1) else versions[0]}"
+        )
+
+    version = FEATURIZER_VERSION_MAP.get(name)
+
+    featurizer = PETMADFeaturizer(version=version, device=device, batch_size=batch_size)
+
+    return featurizer
```

Comment:
I would rather do something like this:

```suggestion
    if name == "pet-mad-1.0":
        device = ...
        model = PETMADFeaturizer(version="1.0.0", device=device, batch_size=1)
        return metatomic_featurizer(model)
    else:
        raise ValueError("unknown featurizer: …")
```